### PR TITLE
Set up Nginx for the REST api

### PR DESCRIPTION
**Pull request by klopyrev**
3 changes, specifically:
1) Get config_gen to run again locally (It was failing on SSL_CERTIFICATE|length).
2) Add the REST server reverse proxy rule.
3) Set up Nginx for local development. I'll add instructions to the Wiki once the code is checked in, but basically:

cl server
cl rest-server
cd codalab-worksheets/codalab
./manage runserver 127.0.0.1:2700

sudo apt-get install nginx
./manage config_gen
sudo cp config/generated/nginx.conf /etc/nginx/sites-enabled
sudo service nginx restart

@percyliang @kashizui @andreweduffy

### CODE REVIEW

**Comment by klopyrev** [comment]
Added another change that makes sure that SSL works. We need to tell Django and Bottle what scheme the request happened with (http vs https). Also, setting X-Forwarded-Host is more appropriate than Host.

**Comment by klopyrev** [comment]
Oh, also, using $http_host vs $host ensures that the port number is included.

File: codalab/manage [hunk 1]
```diff
@@ -4,4 +4,5 @@
 #   runserver 0.0.0.0:8000
 #   syncdb --migrate
 # Add --traceback to print out more.
+source ../venv/bin/activate
```
Comment (percyliang) [inline codalab/manage]: Why is this needed?  Is there an environment variable that needs to be picked up?

**Comment by percyliang** [comment]
Currently, one can (i) run the django app without nginx for local development or (ii) use django with nginx for deployment.  Do we need a separate nginx for local deployment that's not the same as production?  Can we keep things simpler?

**Comment by klopyrev** [comment]
You can't do local development without Nginx. However, the Nginx config
generated by config_gen works, so you just load it up and start Nginx.


Konstantin

**Comment by percyliang** [comment]
Hm, that's unfortunate.  Before, it was nice to be able to fire up a local
instance of CodaLab without root access.  Any possibility of keeping it
that way?

On Sun, Mar 6, 2016 at 5:17 PM, Konstantin Lopyrev <notifications@github.com

**Comment by klopyrev** [comment]
What's the use case for that?


Konstantin

**Comment by percyliang** [comment]
1) Sometimes, users who have an account on a cluster don't have root access
and want to use CodaLab.  CodaLab fundamentally doesn't require root, so we
should be able to not need it.

2) I'm not a big fan of global configuration like:
sudo cp config/generated/nginx.conf /etc/nginx/sites-enabled
This means that you can't run two instances of CodaLab very easily on the
same machine without opening up the hood.

Just as virtualenv nicely allows non-root users to maintain different
Python environment, I think people should be able to try out different
CodaLab versions/environments in the same way.


On Sun, Mar 6, 2016 at 5:24 PM, Konstantin Lopyrev <notifications@github.com

**Comment by klopyrev** [comment]
1) Users can use CodaLab without running Nginx. You can't run your own
version of CodaLab without it.

2) You can set up multiple sites by running Nginx with multiple configs.

You can technically run Nginx without root. It just takes another config
file. You do, however, need to be able to install Nginx.

We can talk more about this on Tuesday. I personally don't see who would
want to run their own version of CodaLab and would not be able to get their
hands on a machine without root access.


Konstantin

File: codalab/manage (hunk 1 above)
Comment (klopyrev) [inline codalab/manage]: Yup, some of the generated config files use the VIRTUAL_ENV variable.

**Comment by klopyrev** [comment]
Is this LGTM?

**Comment by percyliang** [comment]
LGTM
